subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/DirectoryBuildCacheFixture.groovy

```diff
@@ -31,19 +31,31 @@ trait DirectoryBuildCacheFixture {
         settingsFile << buildCache.localCacheConfiguration()
     }
 
+    // Spock 2 executes @Before after the setup() methods
+    // this is a workaround for tests that use this fixture from their setup() methods
+    private void initIfNeeded() {
+        if (buildCache == null) {
```

Comment:
Shouldn't this test be inside `setupCacheDirectory()` so that the method body is not executed twice? Maybe not a problem in practice...